### ISSUE TITLE
Report: Failed posts when editing. 

### ISSUE THREAD

**Issue by aerych** [wordpress-mobile/WordPress-iOS#3045]
We've had several reports in the forums where a user edits an existing post, and the updated post displays a "failed" status in the post list, yet the edit occurred successfully on their site.  Users report that this happens on self-hosted sites with or without the Jetpack plugin enabled.  

I've tried reproducing this with out success. The only weirdness I've noticed is a race condition between publishing the edited post, and refreshing the post list which can result in the old version of the post stored on the device until a subsequent request. (Maybe related to #2804). 

Relevant forum threads: 

https://ios.forums.wordpress.org/topic/media-upload-failure-and-not-posting
https://ios.forums.wordpress.org/topic/post-edit-failures
https://ios.forums.wordpress.org/topic/edited-posts-show-as-failed-in-app-but-update-posts

cc @bummytime

**Comment by aerych** [comment]
Possibly related: https://github.com/wordpress-mobile/wpxmlrpc/issues/20

We found a case where sites hosted at bloggersdelight.dk were experiencing a failure status when publishing posts.  The posts were successfully posted to the blog, but the XML-RPC response from the was being truncated more than the wpxmlrpc API could repair.

**Comment by richiwalt** [comment]
I am experiencing this - however, I've isolated this to only occur on post that have a "featured image" associated to the article.   If a post does not have a "featured image" (even if it has images in it) then the mobile app can make edits without the FAILED error.  But no post with a feature image for the article can be editors without that error occurring.    

Again, the error seems to be bogus - as the intended edits go thru successfully anyway.  The annoyance is that FAILED post float to the top out of chronilogical order and accumulate there.

**Comment by aerych** [comment]
> I've isolated this to only occur on post that have a "featured image" associated to the article

Thanks @richiwalt.  Sounds like there are a few different triggers.  Are you able to debug the XML-RPC calls made by the app?  It would be very helpful to know if you're also experiencing the truncated response.

**Comment by i0n1ca** [comment]
Here is a log from my selfhosted site isay.ro

![image](https://cloud.githubusercontent.com/assets/3650133/6036553/5640f5be-ac4f-11e4-8d52-ffb2c6619874.jpg)
![image](https://cloud.githubusercontent.com/assets/3650133/6036554/56435692-ac4f-11e4-931e-f028403cc611.jpg)

**Comment by richiwalt** [comment]
View the last line of these logs ...

2015/02/04 07:44:04:690  Copying attribute content
2015/02/04 07:44:04:690  Copying attribute date_created_gmt
2015/02/04 07:44:04:691  Copying attribute mt_excerpt
2015/02/04 07:44:04:691  Copying attribute mt_text_more
2015/02/04 07:44:04:691  Copying attribute password
2015/02/04 07:44:04:691  Copying attribute permaLink
2015/02/04 07:44:04:691  Copying attribute postID
2015/02/04 07:44:04:691  Copying attribute postTitle
2015/02/04 07:44:04:691  Copying attribute post_thumbnail
2015/02/04 07:44:04:691  Copying attribute remoteStatusNumber
2015/02/04 07:44:04:691  Copying attribute status
2015/02/04 07:44:04:691  Copying attribute wp_slug
2015/02/04 07:44:04:691  Copying attribute authorAvatarURL
2015/02/04 07:44:04:692  Copying attribute geolocation
2015/02/04 07:44:04:692  Copying attribute latitudeID
2015/02/04 07:44:04:692  Copying attribute longitudeID
2015/02/04 07:44:04:692  Copying attribute postFormat
2015/02/04 07:44:04:692  Copying attribute publicID
2015/02/04 07:44:04:692  Copying attribute tags
2015/02/04 07:44:04:692  Copying relationship blog
2015/02/04 07:44:04:692  Skipping relationship original
2015/02/04 07:44:04:692  Skipping relationship revision
2015/02/04 07:44:04:692  Copying relationship comments
2015/02/04 07:44:04:692  Copying relationship media
2015/02/04 07:44:04:692  Copying relationship categories
2015/02/04 07:44:11:936  post failed: Invalid attachment ID.

**Comment by richiwalt** [comment]
I am guessing for me an attachment ID has something to do with the post image ?   If a post does not have a "Post Image" then it doesn't report a failed status.  Again, for me this only happens when I am editing a post already posted that already has a post image.   The edit does go through - but the failed status occurrs.

**Comment by aerych** [comment]
Thank you @i0n1ca! Thank you @richiwalt!  
The seeing the `invalid attachment ID` in the logs is a great clue.

**Comment by i0n1ca** [comment]
Found another fail on SyncBlogsAccount. Maybe will help. 

![image](https://cloud.githubusercontent.com/assets/3650133/6043897/704054e8-ac98-11e4-8ba2-e9672ae566e7.jpg)
![image](https://cloud.githubusercontent.com/assets/3650133/6043898/70e0f75e-ac98-11e4-987d-bdae156bc78e.jpg)

**Comment by aerych** [comment]
I can reproduce this consistently.  Steps: 

- Create a new post and add a featured image.
- Save the post. 
- Post saves successfully. 
- Edit the post. 
- Change the title or content but do not change the featured image. 
- Save
The post fails with an XML-RPC fault:
```
<methodResponse> 
	<fault> 
		<value> 
			<struct> 
				<member> 
					<name>faultCode</name> 
					<value>
						<int>404</int>
					</value> 
				</member> 
				<member> 
					<name>faultString</name> 
					<value>
						<string>Invalid attachment ID.</string>
					</value> 
				</member> 
			</struct> 
		</value> 
	</fault> 
</methodResponse>
```

It appears the trigger is specifying the same `wp_post_thumbnail` ID already present on the post when saving edits. 

Prior to fef2a3a932434dd4b9da360dfa2ee89a00c85f07 we were [removing this field](https://github.com/wordpress-mobile/WordPress-iOS/blob/09558cc9a2938aa2b0c869ae930f567204cab76c/WordPress/Classes/Models/Post.m#L453) if the image had not changed.

**Comment by aerych** [comment]
Calling it closed with #3186
